### PR TITLE
updated dependencies. sync to 5.0.16 & core to 6.0.19

### PR DESCRIPTION
**Pull request by steffenagger**

### CODE REVIEW

**Comment by kneth** [comment]
Unfortunately we need to update Visual Studio and wait for a new sync release.

**Comment by blagoev** [comment]
also we will need an updated changelog with the fixes in these dependencies @bmunkholm what should we write for these?

**Comment by bmunkholm** [comment]
You would normally copy/paste any relevant entries from the release notes of the dependencies, but rephrasing them to make them relevant to the JS SDK and it's users. That could be referring to what reported bugs it fixes or how the bug fixed would occur for the users. (sometimes it can be an advantage to see what other SDK's did if they already released :-) ).
Kenneth normally does this, so he might have other tips...

**Comment by kneth** [comment]
Core's change log says `If a realm needs upgrade during opening, the program might abort in the "migrate_links" stage` but I think we could formulate it as `When the schema has been updated, could under rare cases crash`. It has only been reported for Cocoa: https://github.com/realm/realm-cocoa/issues/6680

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
+x.x.x Release notes (yyyy-MM-dd)
+=============================================================
+### Enhancements
+* None.
+
+### Fixed
+* <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-js/issues/????), since v?.?.?)
```
Comment (blagoev) [inline CHANGELOG.md:7]: this line should be removed before the release.

**Review by blagoev — APPROVED** [review]
just remove the guidance line

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+* Rare crash when a schema was updated ([#6680](https://github.com/realm/realm-cocoa/issues/6680))
+* Rare crash when updating Realm file from v9 to v10. ([#3836](https://github.com/realm/realm-core/issues/3836))
```
Comment (kneth) [inline CHANGELOG.md:8]:
```suggestion
* Rare crash when updating Realm file from v9 to v10. ([realm/realm-core#3836](https://github.com/realm/realm-core/issues/3836))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,24 @@
+x.x.x Release notes (yyyy-MM-dd)
+=============================================================
+### Enhancements
+* None.
+
+### Fixed
+* <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-js/issues/????), since v?.?.?)
+* Rare crash when a schema was updated ([#6680](https://github.com/realm/realm-cocoa/issues/6680))
```
Comment (kneth) [inline CHANGELOG.md:7]:
```suggestion
* Rare crash when a schema was updated ([realm/realm-cocoa#6680](https://github.com/realm/realm-cocoa/issues/6680))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+* Rare crash when updating Realm file from v9 to v10. ([#3836](https://github.com/realm/realm-core/issues/3836))
+* Upgrading a table with only backlink columns could result in a crash.
```
Comment (kneth) [inline CHANGELOG.md:9]: "backlink columns" -> maybe "linkingObjects properties"

**Review by kneth — APPROVED** [review]
Minor comments

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,23 @@
+x.x.x Release notes (yyyy-MM-dd)
+=============================================================
+### Enhancements
+* None.
+
+### Fixed
+* Rare crash when a schema was updated ([realm/realm-cocoa#6680](https://github.com/realm/realm-cocoa/issues/6680))
```
Comment (bmunkholm) [inline CHANGELOG.md:7]: Normally we should aim to describe how the user can see observe the crash so it's easy for them to see if this fixes something they have seen. E.g. here I would just add "Rare crash (in realm::Table::migrate_links)". But I assume we haven't had any JS users experience this?

**Pull request merged**